src/Maps/MapData.h

```diff
@@ -87,9 +85,10 @@ class MapData
 	uint32_t mapTileHeight;
 
 	std::size_t GetTileIndex(std::size_t x, std::size_t y) const;
-	uint32_t GetWidthInTilesLog2(uint32_t widthInTiles) const;
 
 	// Write
+	MapHeader CreateHeader() const;
```

Comment:
Random aside: I'm a little surprised this worked after changing `MapHeader` to a forward declare and removing the include. It's not used as a pointer here, at least not syntactically, though in terms of implementation details, yes, it will pass a pointer to write the return value to.

I suspect this will require the caller to see the full definition of `MapHeader`. It's now a private method though, so most code won't be able to call this method.

Very interesting solution.